libraries/launcher/org/multimc/EntryPoint.java

```diff
@@ -16,136 +16,123 @@
 
 import org.multimc.onesix.OneSixLauncher;
 
-import java.io.*;
-import java.nio.charset.Charset;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 public class EntryPoint
 {
-    private enum Action
-    {
-        Proceed,
-        Launch,
-        Abort
-    }
+
+    private static final Logger LOGGER = Logger.getLogger("EntryPoint");
+
+    private final ParamBucket params = new ParamBucket();
+
+    private org.multimc.Launcher launcher;
 
     public static void main(String[] args)
     {
         EntryPoint listener = new EntryPoint();
+
         int retCode = listener.listen();
+
         if (retCode != 0)
         {
-            System.out.println("Exiting with " + retCode);
+            LOGGER.info("Exiting with " + retCode);
+
             System.exit(retCode);
         }
     }
 
     private Action parseLine(String inData) throws ParseException
     {
-        String[] pair = inData.split(" ", 2);
+        String[] pair = inData.split("\\s+", 2);
 
-        if(pair.length == 1)
-        {
-            String command = pair[0];
-            if (pair[0].equals("launch"))
+        if (pair.length == 0)
+            throw new ParseException("Unexpected empty string!");
+
+        switch (pair[0]) {
+            case "launch": {
                 return Action.Launch;
+            }
 
-            else if (pair[0].equals("abort"))
+            case "abort": {
                 return Action.Abort;
+            }
 
-            else throw new ParseException("Error while parsing:" + pair[0]);
-        }
+            case "launcher": {
+                if (pair.length != 2)
+                    throw new ParseException("Expected 2 tokens, got 1!");
```

Comment:
this will also be thrown if there are three or more tokens

Reply:
There are second argument in split method, which forces to split string only on two parts.

Reply:
I think I'll change it to output pair.length as amount of tokens, just for the clarity

Reply:
> There are second argument in split method, which forces to split string only on two parts.

Oh, you are right. Might have missed that 😅